compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/window/ComposeWindow.js.kt

```diff
@@ -134,75 +174,80 @@ private class ComposeWindow(
+            layer.view.onPointerEvent(event.toSkikoEvent(SkikoPointerEventKind.DOWN))
         }
 
-        canvas.addTypedEvent<MouseEvent>("mouseup") { event, skikoView ->
+        addTypedEvent<MouseEvent>("mouseup") { event ->
             isPointerPressed = false
-            skikoView.onPointerEvent(event.toSkikoEvent(SkikoPointerEventKind.UP))
+            layer.view.onPointerEvent(event.toSkikoEvent(SkikoPointerEventKind.UP))
         }
 
-        canvas.addTypedEvent<MouseEvent>("mousemove") { event, skikoView ->
+        addTypedEvent<MouseEvent>("mousemove") { event ->
             if (isPointerPressed) {
-                skikoView.onPointerEvent(event.toSkikoDragEvent())
+                layer.view.onPointerEvent(event.toSkikoDragEvent())
             } else {
-                skikoView.onPointerEvent(event.toSkikoEvent(SkikoPointerEventKind.MOVE))
+                layer.view.onPointerEvent(event.toSkikoEvent(SkikoPointerEventKind.MOVE))
             }
         }
 
-        canvas.addTypedEvent<WheelEvent>("wheel") { event, skikoView ->
-            skikoView.onPointerEvent(event.toSkikoScrollEvent())
+        addTypedEvent<WheelEvent>("wheel") { event ->
+            layer.view.onPointerEvent(event.toSkikoScrollEvent())
         }
 
         canvas.addEventListener("contextmenu", { event ->
             event.preventDefault()
         })
 
-        canvas.addTypedEvent<KeyboardEvent>("keydown") { event, skikoView ->
-            val processed = skikoView.onKeyboardEventWithResult(event.toSkikoEvent(SkikoKeyboardEventKind.DOWN))
+        addTypedEvent<KeyboardEvent>("keydown") { event ->
+            val processed = layer.view.onKeyboardEventWithResult(event.toSkikoEvent(SkikoKeyboardEventKind.DOWN))
             if (processed) event.preventDefault()
         }
 
-        canvas.addTypedEvent<KeyboardEvent>("keyup") { event, skikoView ->
-            val processed = skikoView.onKeyboardEventWithResult(event.toSkikoEvent(SkikoKeyboardEventKind.UP))
+        addTypedEvent<KeyboardEvent>("keyup") { event ->
+            val processed = layer.view.onKeyboardEventWithResult(event.toSkikoEvent(SkikoKeyboardEventKind.UP))
             if (processed) event.preventDefault()
         }
-
-        initMediaEventListener {
-            resize(getParentContainerBox())
-        }
     }
 
     init {
         initEvents(canvas)
+        state.init()
 
         canvas.setAttribute("tabindex", "0")
 
         layer.setDensity(density)
         layer.setContent {
             CompositionLocalProvider(
                 LocalSystemTheme provides systemThemeObserver.currentSystemTheme.value,
-                content = content
+                content = {
+                    content()
+                    rememberCoroutineScope().launch {
+                        state.sizeFlow().collect { size ->
+                            this@ComposeWindow.resize(size)
+                            delay(100)
```

Comment:
We have this delay. I'm not sure it would work fine with a flow created for a custom size provider (Current `private fun(suspend () -> IntSize).asComposeWindowState(): ComposeWindowState`)

The DefaultWindowState uses a Channel with Capacity=1 and Drop Oldest policy. The delay doesn't break it, because Drop Oldest makes sure the most recent IntSize is always in the Channel. 
This approach helps to not be overloaded by a lot of events.

But in case of a custom resize and a flow, I guess our handler would have to process all resize events. And delay would make it even longer. I think we need to skip some events.
I might be mistaken though in this aspect.
___

Maybe it's okay to remove this delay for now. It works not so bad on my laptop without this delay in Default case.

Reply:
@eymar I removed delay but if we'll need it we'll be able to reintroduce it non-invasively, without affecting custom `requestSize`